BatchBrainMRTumorSegmentation

saimasafdar2021 opened the pull request:
does tumor segmentation for a batch using T1, T1c, flair and T2 images. It uses the brats toolkit to perform segmentation. Its a wrapper for brats toolkit for performing batch processing.

<!--
Thank you for contributing to 3D Slicer!
- To add a new extension with this pull request: Please keep content of "New extension" section and put an 'x' in the brackets for each todo item to indicate that you have accomplished that prerequisite.
- To update an existing extension with this pull request: Please delete all text in this template and just describe which extension is updated and optionally tell us in a sentence what has been changed. To make extension updates easier in the future you may consider replacing specific git hash in your s4ext file by a branch name (for example: `main` for Slicer Preview Releases; `(majorVersion).(minorVersion)` such as `4.10` for Slicer Stable Releases).
-->

# New extension

<!-- To make sure users can find your extension, understand what it is intended for and how to use it, please complete the checklist below. You do not need to complete all the item by the time you submit the pull request, but most likely the changes will only be merged if all the tasks are done. See more information about the submission process here: https://slicer.readthedocs.io/en/latest/developer_guide/extensions.html -->

- [ ] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [x] Repository name is Slicer+ExtensionName
- [x] Repository is associated with `3d-slicer-extension` GitHub topic so that it is listed [here](https://github.com/topics/3d-slicer-extension). To edit topics, click the settings icon in the right side of "About" section header and enter `3d-slicer-extension` in "Topics" and click "Save changes". To learn more about topics, read https://help.github.com/en/articles/about-topics
- [x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x] Any known related patents must be mentioned in the extension description.
- [x] LICENSE.txt is present in the repository root. MIT  (https://choosealicense.com/licenses/mit/) or Apache (https://choosealicense.com/licenses/apache-2.0/) license is recommended. If source code license is more restrictive for users than MIT, BSD, Apache, or 3D Slicer license then the name of the used license must be mentioned in the extension description.
- [x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (main, release, ...) instead of a specific git hash to avoid re-submitting pull request whenever the extension is updated
- [x] Extension icon URL is correct (do not use the icon's webpage but the raw data download URL that you get from the download button - it should look something like this: https://raw.githubusercontent.com/user/repo/main/SomeIcon.png)
- [x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [x] Homepage URL points to valid webpage containing the following:
  - [x] Extension name
  - [x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [x] Description of contained modules: at one sentence for each module
  - [x] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [x] Publication: link to publication and/or to PubMed reference (if available)
  - [x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/main/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work. Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.
  - [x] Content of submitted s4ext file is consistent with the top-level CMakeLists.txt file in the repository (description, URLs, dependencies, etc. are the same)
- Hide unused features in the repository to reduce noise/irrelevant information:
  - [x] Click `Settings` and in repository settings uncheck `Wiki`, `Projects`, and `Discussions` (if they are currently not used)
  - [x] Click the settings icon next to `About` in the top-right corner of the repository main page and uncheck `Releases` and `Packages` (if they are currently not used)
<!-- If you have any questions or comments then please describe them here. -->

saimasafdar2021 commented:
Hi, 
Is there anything on my side to do for publishing it with 3D Slicer. When I can see it in the slicer extension wizard? I have written a publication regarding this extension which requires it to be published first. We are sending the Publication to SoftwareX journal. Could you please send update on this publishing of this extension. 

Thank you

regards,
Saima

pieper commented:
I didn't try the code, but a few small notes from a quick look:

* You say brats needs to be installed in a different python but you don't give any details about it.  This isn't a common way to distribute software for slicer so some information for users would help.  Does that other tool require a GPU?  Does it work on all platforms?  You need to provide links to the required software.
* You have some extra python files in the same directory with you scripted module source code.  Those will probably be loaded to test for modules and generate warnings.  It's not clear from a quick look what those do so they should be removed or moved.
* There is a __pycache__ folder in your repo that should probably not be in git.

saimasafdar2021 commented:
Hi, 
I have made the changes required.
I have added the link to install the brats-toolkit and other installations suchas nvidia.
I have removed the folder pycache
I have made the scripted files to another folder. These files are required by the program from within the module. 

regards,
Saima

pieper commented:
Does anyone else want to have a look at this before merging? @lassoan @jcfr ?

saimasafdar2021 commented:
Please recommend any changes if required.

many thanks.

regards,
Saima

saimasafdar2021 commented:
Hi,
I am writing to know the status of this extension. Please let me know if I need to change anything to get it published. 

regards,
Saima

lassoan commented:
Sorry for the delayed feeback.

The extension cannot be built or packaged, because it does not follow the extension template. The most important thing is that the top-level CMakeLists.txt file is missing!!

You can use the Extension Wizard module in Slicer to create a valid extension template and add those files to your repository.

Also note that the .s4ext file is corrupted (`contributors` field is missing, field name of `description` field is missing, there may be other issues as well). I would recommend to [build 3D Slicer](https://slicer.readthedocs.io/en/latest/developer_guide/build_instructions/linux.html) and then build your extension to test that it can be built and packaged successfully and then you also get an .s4ext file in your build tree that you can use in this pull request (just modify `scmrevision` to `main`). If you are unable to build Slicer then you must really, really carefully copy all the content from any other extension's s4ext file and update it to match the contents of the top-level CMakeLists.txt file in your repository.

saimasafdar2021 commented:
Hi Andras,
I have made the changes you suggested. Carefully copied the .s4ext file and checked all required fields. I have created the cmakeList.txt file and is consistent with the contents in the .s4ext file. 

Please let me know if there are any other requirements. 

thank you so much.

regards,
Saima

jcfr commented:
Unable to force push changes onto [UWA-Medical-Physics/ExtensionsIndex@main](https://github.com/UWA-Medical-Physics/ExtensionsIndex/tree/main), the updated changes are published here https://github.com/Slicer/ExtensionsIndex/compare/main...jcfr:ExtensionsIndex:UWA-Medical-Physics-main

Consider running the following command to update this pull request:

```
git clone git@github.com:UWA-Medical-Physics/ExtensionsIndex.git
cd ExtensionsIndex
git checkout main
git remote add jcfr https://github.com/jcfr/ExtensionsIndex.git
git fetch jcfr
git reset --hard jcfr/UWA-Medical-Physics-main
git push origin main --force
```

jcfr commented:
Consider address the following issue:
* https://github.com/UWA-Medical-Physics/SlicerBatchBrainMRTumorSegmentation/issues/2

saimasafdar2021 commented:
Hi Jean,
I have successfully submitted the extension as asked. 
 Create SlicerBrats.json #2069 


regards,
Saima

saimasafdar2021 commented:
> Consider address the following issue:
> 
>     * [Ensure bratsFusionator/bratsPreprocessor/bratsSegmentor are distributed UWA-Medical-Physics/SlicerBatchBrainMRTumorSegmentation#2](https://github.com/UWA-Medical-Physics/SlicerBatchBrainMRTumorSegmentation/issues/2)

Hi Jean,
I have successfully submitted the extension as asked.
Create SlicerBrats.json https://github.com/Slicer/ExtensionsIndex/pull/2069

regards,
Saima

jcfr commented:
Please, consider re-basing this topic against the latest `main`

jcfr commented:
> Please, consider re-basing this topic against the latest `main`

:pray:

jcfr commented:
> Please, consider re-basing this topic against the latest `main`

:pray:
